scholly/base/src/index.js

```diff
@@ -7,11 +7,69 @@ import 'bootstrap/dist/css/bootstrap.css';
 import 'bootstrap/dist/css/bootstrap-grid.css';
 import 'bootstrap/dist/css/bootstrap-reboot.css';
 
-import App from './components/App';
+import { LAYOUT_OPTION } from './global/data';
+
+/// Decides which of the root App components of the various layouts to display.
+/// This determination is based off the provided LAYOUT_OPTION environment variable.
+const layoutToDisplay = () => {
+  switch (LAYOUT_OPTION) {
+    case 1: // TODO: there might be a better way to do this.
+      // eslint-disable-next-line no-case-declarations
+      const AppLayout1 = React.lazy(() => import('./layouts/layout1/App'));
+      return (
+        <>
+          <React.Suspense fallback={<></>}>
+            <AppLayout1 />
+          </React.Suspense>
+        </>
+      );
+    case 2:
+      // eslint-disable-next-line no-case-declarations
+      const AppLayout2 = React.lazy(() => import('./layouts/layout2/App'));
+      return (
+        <>
+          <React.Suspense fallback={<></>}>
+            <AppLayout2 />
+          </React.Suspense>
+        </>
+      );
+    case 3:
+      // eslint-disable-next-line no-case-declarations
+      const AppLayout3 = React.lazy(() => import('./layouts/layout3/App'));
+      return (
+        <>
+          <React.Suspense fallback={<></>}>
+            <AppLayout3 />
+          </React.Suspense>
+        </>
+      );
+    case 4:
+      // eslint-disable-next-line no-case-declarations
+      const AppLayout4 = React.lazy(() => import('./layouts/layout4/App'));
+      return (
+        <>
+          <React.Suspense fallback={<></>}>
+            <AppLayout4 />
+          </React.Suspense>
+        </>
+      );
+
+    default:
+      // eslint-disable-next-line no-case-declarations
+      const DefaultAppLayout = React.lazy(() => import('./layouts/layout1/App'));
+      return (
+        <>
+          <React.Suspense fallback={<></>}>
+            <DefaultAppLayout />
+          </React.Suspense>
+        </>
+      );
+  }
+};
```

Comment:
Can this be tested E2E to check if this works correctly. I know that it shows up in debug mode, but this lazy loading may not work correctly if we're doing a production build.

Reply:
This issue will be resolved on a new PR by @kdav108